File: visionflow/frontend/src/types/agent.ts
export type AgentState =
  | 'offline'
  | 'online'
  | 'recording'
  | 'replaying'
  | 'paused_conflict'
  | 'waiting_hitl';

export interface AgentStatus {
  agent_state: AgentState;
  last_command_id: string | null;
  recording_session_id: string | null;
  recording_event_count: number;
  recording_elapsed: number;
  replay_run_id: string | null;
  replay_workflow_id: string | null;
  replay_current_step: number;
  replay_total_steps: number;
  replay_step_description: string | null;
  error: string | null;
}

export type CommandType =
  | 'start_recording'
  | 'pause_recording'
  | 'resume_recording'
  | 'stop_recording'
  | 'start_run'
  | 'abort_run'
  | 'hitl_response'
  | 'ping';

File: visionflow/backend/backend/schemas/agent.py
"""Agent command/status schemas."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel

from shared.enums import AgentState, CommandType


class AgentCommandRequest(BaseModel):
    type: CommandType
    payload: dict[str, Any] = {}


class AgentStatusResponse(BaseModel):
    agent_state: AgentState = AgentState.OFFLINE
    last_command_id: str | None = None
    recording_session_id: str | None = None
    recording_event_count: int = 0
    recording_elapsed: float = 0
    replay_run_id: str | None = None
    replay_workflow_id: str | None = None
    replay_current_step: int = 0
    replay_total_steps: int = 0
    replay_step_description: str | None = None
    error: str | None = None
